packages/decoder/lib/decoders.ts

```diff
@@ -133,6 +144,146 @@ export class ProjectDecoder {
     debug("done with allocation");
   }
 
+  /**
+   * **This function is asynchronous.**
+   *
+   * Adds compilations to the decoder after it has started.  Note it is
+   * only presently possible to do this with a `ProjectDecoder` and not
+   * with the other decoder classes.
+   *
+   * @param compilations The compilations to be added.  Take care that these
+   * have IDs distinct from those the decoder already has.
+   */
+  public async addCompilations(
+    compilations: Compilations.Compilation[]
+  ): Promise<void> {
+    //first: make sure we're not adding a compilation with an existing ID
+    const existingIds = new Set(
+      this.compilations.map(compilation => compilation.id)
+    );
+    //we use a find() rather than a some() so that we can put the ID in the error
```

Comment:
There could be more conflicts. Maybe we share them all in the error?

Reply:
Oh, hm, yeah, that's a thing we could do.  It's more trouble though... do you think that's worth it?

Reply:
Yes, I think it's best to give full information feedback when we can.

Reply:
OK, I've updated it so that reporting here is now based around arrays and sets of IDs, instead of just stopping as soon as there's a single conflict.